### ISSUE TITLE
Detailing EDHOC error in CoAP error

### ISSUE THREAD

**Issue by gselander** [lake-wg/edhoc#134]
New section A.3.1 (see below) has this requirement:

"EDHOC error messages sent as CoAP responses MUST be error
   responses"

1. Does all CoAP error responses support payload? 
2. Should it not be stated that they MUST contain the EDHOC message as payload? 
3. Should Content-Format be included in those messages?

"  When using EDHOC over CoAP for establishing an OSCORE Security
   Context, EDHOC error messages sent as CoAP responses MUST be error
   responses, i.e., they MUST specify a CoAP error response code.  In
   particular, it is RECOMMENDED that such error responses have response
   code either 4.00 (Bad Request) in case of client error (e.g., due to
   a malformed EDHOC message), or 5.00 (Internal Server Error) in case
   of server error (e.g., due to failure in deriving EDHOC key
   material)."

**Comment by gselander** [comment]
@marco-tiloca-sics Maybe you have a proposal?

**Comment by marco-tiloca-sics** [comment]
> Does all CoAP error responses support payload?

Yes, see https://datatracker.ietf.org/doc/html/rfc7252/#section-5.9.2  and https://datatracker.ietf.org/doc/html/rfc7252/#section-5.9.3 . Then, if content-format is not specified, the possible payload is a diagnostic payload encoded in UTF-8 --- see https://datatracker.ietf.org/doc/html/rfc7252/#section-5.5.2 --- which leads to the following point.

> Should it not be stated that they MUST contain the EDHOC message as payload?

Yes, for consistency with the text earlier in Appendix A.3, e.g., "EDHOC message_2 or the EDHOC error message is sent from the server to the client in the payload of ...".

> Should Content-Format be included in those messages?

Actually yes, as an explicit hint to the client that the payload is not a generic diagnostic payload encoded in UTF-8, or any other payload originated by an error source different than EDHOC processing.

If this is phrased using "MUST", then the second from last paragraph in https://datatracker.ietf.org/doc/html/draft-ietf-core-oscore-edhoc-01#section-3.3  can be shortened, since that document would inherit also this behavior as already defined in the EDHOC document.

**Comment by emanjon** [comment]
Seems to me that this can be closed now
